src/devices/bus/coco/coco_pak.cpp

```diff
@@ -257,3 +261,212 @@ void coco_pak_banked_device::scs_write(offs_t offset, u8 data)
+//-------------------------------------------------
+
+coco_pak_rmulticart_device::coco_pak_rmulticart_device(const machine_config &mconfig, device_type type, const char *tag, device_t *owner, u32 clock)
+	: coco_pak_device(mconfig, type, tag, owner, clock)
+	//, m_pos(0)
+{
+}
+coco_pak_rmulticart_device::coco_pak_rmulticart_device(const machine_config &mconfig, const char *tag, device_t *owner, u32 clock)
+	: coco_pak_rmulticart_device(mconfig, COCO_PAK_RMULTICART, tag, owner, clock)
+{
+}
+
+//-------------------------------------------------
+//  rom_region - device-specific ROM region
+//-------------------------------------------------
+
+const tiny_rom_entry *coco_pak_rmulticart_device::device_rom_region() const
+{
+	return ROM_NAME( coco_pak_rmulticart );
+}
+
+//-------------------------------------------------
+//  device_start - device-specific startup
+//-------------------------------------------------
+
+void coco_pak_rmulticart_device::device_start()
+{
+	coco_pak_device::device_start();
+
+	save_item(NAME(Page));
+	save_item(NAME(CTRL));
+	save_item(NAME(PageSize));
+	save_item(NAME(PageMask));
+}
+
+//-------------------------------------------------
+//  device_reset - device-specific startup
+//-------------------------------------------------
+
+void coco_pak_rmulticart_device::device_reset()
+{
+	Page = 0;
+	CTRL = RMULTI_AUTO;
+	PageSize = 0x2000;
+	PageMask = RMULTI_PAGE_MASK8;
+
+	SetCART();
+}
+
+//-------------------------------------------------
+//  get_cart_base
+//-------------------------------------------------
+
+u8 *coco_pak_rmulticart_device::get_cart_base()
+{
+	return m_eprom->base() + (Page * PageSize);
+}
+
+//-------------------------------------------------
+//  get_cart_size
+//-------------------------------------------------
+
+u32 coco_pak_rmulticart_device::get_cart_size()
+{
+	return PageSize;
+}
+
+//-------------------------------------------------
+//  cts_read
+//-------------------------------------------------
+
+u8 coco_pak_rmulticart_device::cts_read(offs_t offset)
+{
+	return m_eprom->base()[(Page * PageSize) + (offset % PageSize)];
+}
+
+// Set CART line dependent on CTRL reg & Autostart input.
+void coco_pak_rmulticart_device::SetCART()
+{
+	if (m_cart->exists())
+	{
+		auto cart_line = (m_autostart.read_safe(0x01) && (CTRL & RMULTI_AUTO))
+			? line_value::Q
+			: line_value::CLEAR;
+
+		// normal CoCo PAKs tie their CART line to Q - the system clock
+		set_line_value(line::CART, cart_line);
+	}
+}
+
+//-------------------------------------------------
+//  scs_read
+//-------------------------------------------------
+
+u8 coco_pak_rmulticart_device::scs_read(offs_t offset)
+{
+	switch(offset)
+	{
+		// $FF50
+		case 0x10:
+			logerror("Read Page: %02X\n",Page);
+			return Page;
+			break;
+		
+		// $FF52
+		case 0x12:
+			logerror("Read CTRL: %02X\n",CTRL);
+			return (RMULTI_RS << 4) | (CTRL & 0x0F);
+			break;
+			
+		default:
+			return 0xFF;
+			break;
```

Comment:
You don't need to `break `after a `return `statement, additionally I'd ditch the default label and just move the `return 0xff `as function closure instead.